### PR TITLE
Pin Pygments & Jinja to specific versions

### PR DESCRIPTION
**Pull request by justinmayer**
Tests were failing on developer workstations as functional test output became out-of-sync as new versions of Pygments were released. Tests were passing in automated CI, however, because the version was pinned in CI-specific configuration. This moves that pinned dependency specification to a requirements file in order to (hopefully) ensure that test output and Pygments are kept in sync across environments.

Refs #1916 #1961 #2712

### CODE REVIEW

File: pelican/tests/output/basic/a-markdown-powered-article.html [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
 <body id="index" class="home">
         <header id="banner" class="body">
-                <h1><a href="/">A Pelican Blog </a></h1>
```
Comment (avaris) [inline pelican/tests/output/basic/a-markdown-powered-article.html:12]: This is a mysterious change.

Comment (justinmayer) [reply pelican/tests/output/basic/a-markdown-powered-article.html:12]: Agreed. It makes one wonder why the space was there in the first place, no? Perhaps a bug that was fixed?

Comment (justinmayer) [reply pelican/tests/output/basic/a-markdown-powered-article.html:12]: It seems that Jinja v2.11+ is what changed the handling of this particular space.

**Comment by justinmayer** [comment]
Many thanks for reviewing, @avaris.

**Pull request merged**